File: CSharp/modelo/Conta.cs
using QualquerBank.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QualquerBank.modelo
{
    public abstract class Conta
    {
        public static int TotalDeContas {get; private set;}
        public int Numero { get; set; }
        public int Agencia { get; set; }
        public double Saldo { get; protected set; }

        public Conta(int numero, int agencia)
        {
            this.Numero = numero;
            this.Agencia = agencia;
            this.Saldo = 0;
            Conta.TotalDeContas++;
        }

        public virtual bool Sacar(double valor)
        {
            if (valor > 0 && valor <= Saldo)
            {
                Saldo -= valor;
                return true;
            }
            else
            {
                throw new ArgumentException("Valor invalido");
            }
        }
        public bool Depositar(double valor)
        {
            if (valor > 0)
            {
                Saldo += valor;
                return true;
            }

            return false;
        }
        public bool Transferir(double valor, Conta favorecido)
        {
            if (Sacar(valor))
            {
                favorecido.Depositar(valor);
            }

            return false;
        }

        public abstract void ExibeConta();
    }
}

File: Java/contas/Conta.java
package contas;

public abstract class Conta {
	private static int totalDeContas = 0;
	private int numero;
	private int agencia;
	protected double saldo;

	public Conta(int numero, int agencia) {
		this.numero = numero;
		this.agencia = agencia;
		this.saldo = 0;
		Conta.totalDeContas++;
	}

	public int getNumero() {
		return numero;
	}

	public void setNumero(int numero) {
		this.numero = numero;
	}

	public int getAgencia() {
		return agencia;
	}

	public void setAgencia(int agencia) {
		this.agencia = agencia;
	}

	public double getSaldo() {
		return saldo;
	}

	public static int getTotalDeContas() {
		return Conta.totalDeContas;
	}

	public boolean sacar(double valor) {
		if (valor > 0 && valor <= getSaldo()) {
			saldo -= valor;
			return true;
		} else {
			throw new IllegalArgumentException("Valor invalido");
		}
	}

	public boolean depositar(double valor) {
		if (valor > 0) {
			saldo = saldo + valor;
			return true;
		} else {
			throw new IllegalArgumentException("Valor invalido.");
		}
	}

	public boolean transferir(double valor, Conta favorecido) {
		if (sacar(valor) == true) {
			favorecido.depositar(valor);
			return true;
		}

		return false;
	}
	
	public abstract void exibeConta();
}
